torch/backends/cuda/__init__.py

```diff
@@ -310,17 +310,26 @@ def sdp_kernel(
     enable_flash: bool = True,
     enable_math: bool = True,
     enable_mem_efficient: bool = True,
+    backend: Optional[SDPBackend] = None,
```

Comment:
nit: docs for the new arg

Reply:
If you look at v1 of this PR I raised a "warning" saying that  the signature for this function is deprecated and going forward we would only allow for the backend kwarg. I can add that bag if we like, Tbh I think that the prominent use case is for people to select Flash and I want to make that easier

Reply:
devils advocate Q: if the prominent use case is for people to select flash, what's wrong with `sdp_kernel(enable_flash=True)` that they can do today?

Reply:
to enable flash attention and only flash attention they need to do:
`sdp_kernel(enable_math=False, enable_mem_efficient=False)`. This function was originally created for testing but has become popular for people to ensure that they are in fact using Flashattention. I think making the selection be by selecting the kernel explicitly as opposed to turning off the negation was just bad design. Especially if we add new backends, CUDNN people will then need to negate more which feels just wrong

Reply:
cool, I agree that backend is more scalable. Is it out of the question to introduce e.g. `sdp_backend(backend=...)` as a *new* context manager? it could be easier to deprecate the old stuff if you can deprecate the entire thing